pyccel/codegen/printing/ccode.py

```diff
@@ -1295,7 +1307,7 @@ def _print_Constant(self, expr):
 
     def _print_Return(self, expr):
         code = ''
-        args = [VariableAddress(a) if self.stored_in_c_pointer(a) else a for a in expr.expr]
+        args = [VariableAddress(a) if isinstance(a, Variable) and self.stored_in_c_pointer(a) else a for a in expr.expr]
```

Comment:
are sure that we need here to check that the `a` is instance of `variable`, we have this check in the function `stored_in_c_pointer`?

Reply:
We no longer have this check in `stored_in_c_pointer` as `Variable`s are not the only objects that can be stored in pointers. See e27b65dc3d48c52186d12125609cf7017a01e6eb

Reply:
Ah now I see, thanks